src/generator-attribute-manager.cs

```diff
@@ -269,3 +322,261 @@ static bool IsSubclassOf (Type base_class, Type derived_class)
+	{
+		object [] oldCtorValues = HarvestOldAttributeValues (attribute);
+		object [] ctorValues;
+		System.Type [] ctorTypes;
+		FormatNewStyleArguments (oldCtorValues, platformName, out ctorValues, out ctorTypes);
+		return AttributeFactory.CreateNewAttribute (AttributeFactory.IntroducedAttributeType, ctorTypes, ctorValues);
+	}
+}
+
+
+static class AttributeFactory
+{
+	static System.Type PlatformEnum = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.PlatformName");
+	static System.Type PlatformArch = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.PlatformArchitecture");
+
+	public static System.Type IntroducedAttributeType = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.IntroducedAttribute");
+	public static System.Type UnavailableAttributeType = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.UnavailableAttribute");
+	public static System.Type ObsoletedAttributeType = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.ObsoletedAttribute");
+	public static System.Type DeprecatedAttributeType = typeof (TypeManager).Assembly.GetType ("XamCore.ObjCRuntime.DeprecatedAttribute");
+
+	public static System.Attribute CreateNewAttribute (System.Type attribType, System.Type [] ctorTypes, object [] ctorValues)
+	{
+		var ctor = attribType.GetConstructor (ctorTypes);
+		if (ctor == null)
+			throw ErrorHelper.CreateError (1058, "Internal error: could not find a constructor for the mock attribute '{0}'. Please file a bug report (https://bugzilla.xamarin.com) with a test case.", attribType.FullName);
+
+		return (System.Attribute) ctor.Invoke (ctorValues);
+	}
+
+	static Attribute CreateMajorMinorAttribute (System.Type type, PlatformName platform, int major, int minor, byte arch, string message)
+	{
+		var ctorValues = new object [] { (byte)platform, major, minor, arch, message };
+		var ctorTypes = new System.Type [] { PlatformEnum, typeof (int), typeof (int), PlatformArch, typeof (string) };
+		return CreateNewAttribute (type, ctorTypes, ctorValues);
+	}
+
+	public static System.Attribute CreateNewIntroducedAttribute (PlatformName platform, int major, int minor, byte arch = 0xff, string message = "")
+	{
+		return CreateMajorMinorAttribute (IntroducedAttributeType, platform, major, minor, arch, message);
+	}
+
+	public static System.Attribute CreateObsoletedAttribute (PlatformName platform, int major, int minor, byte arch = 0xff, string message = "")
+	{
+		return CreateMajorMinorAttribute (ObsoletedAttributeType, platform, major, minor, arch, message);
+	}
+
+	public static System.Attribute CreateDeprecatedAttribute (PlatformName platform, int major, int minor, byte arch = 0xff, string message = "")
+	{
+		return CreateMajorMinorAttribute (DeprecatedAttributeType, platform, major, minor, arch, message);
+	}
+
+	public static System.Attribute CreateUnavilableAttribute (PlatformName platformName, byte arch = 0xff, string message = "")
```

Comment:
This is the original misspelled method for Sebastian's comment about misspelled Unavailable